rules-reviewed/openshift/mail.windup.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0"?>
+<ruleset id="mail"
+    xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This is a ruleset for detecting Mail API usage when migrating an application to a cloud environment.
+        </description>
+        <dependencies>
+          <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final"/>
+          <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final"/>
+        </dependencies>
+        <sourceTechnology id="java"/>
+        <sourceTechnology id="java-ee"/>
+        <targetTechnology id="cloud-readiness"/>
+        <tag>mail</tag>
+    </metadata>
+    <rules>
+        <rule id="mail-00000">
+            <when>
+                    <javaclass references="javax.mail.{*}">
+                        <location>CONSTRUCTOR_CALL</location>
+                        <location>IMPORT</location>
+                        <location>INHERITANCE</location>
+                        <location>VARIABLE_DECLARATION</location>
+                    </javaclass>
+            </when>
+            <perform>
+              <hint title="JavaMail: direct access of an SMTP server" effort="3" category-id="cloud-optional">
```

Comment:
@jsight This is about outbound mail server connectivity, not the SMTP protocol. This is why I suggested to reformulating without "SMTP".

We should not recommend switching to a dedicated REST service for mail delivery. It might be an interesting idea for specific applications/projects. However, it is not a recommendation that can be done generally. 

Here we detect the usage of backing service (https://12factor.net/backing-services). As mentioned by Marek, it is perfectly possible to use outbound mail connections with EAP on OCP. The point here is probably more to check that the outbound mail connection is defined as a proper backing service.